pruner/pruner.go

```diff
@@ -0,0 +1,41 @@
+package pruner
+
+import (
+	"context"
+	"time"
+
+	"github.com/celestiaorg/celestia-node/header"
+)
+
+// AvailabilityWindow defines an interface for pruning historical data.
+type AvailabilityWindow interface {
+	// IsWithinAvailabilityWindow returns true if the given header is within the
+	// "availability" (sampling) window of the network.
+	IsWithinAvailabilityWindow(eh *header.ExtendedHeader, window time.Duration) bool
```

Comment:
Looks like there is no need to pass window var since it is constant. It can be part of the implementation and accessed locally.

Reply:
I think I did this before for testing purposes (if we wanted to reduce the window / change the window somehow) but i'm happy to remove it and just make it a var if needed.